File: 백준/Silver/15651. N과 M （3）/N과 M （3）.py
def recur(n, m, li):
    if len(li) == m:
        print(" ".join([str(i) for i in li]))
        return
    
    for i in range(1, n + 1):
        li.append(i)
        recur(n, m, li)
        li.pop()

def main(n, m):
    recur(n, m, [])
    
n, m = map(int, input().split())
main(n, m)

File: 백준/Silver/15651. N과 M （3）/N과 M （3）.js
const fs = require('fs');
const [n, k] = fs.readFileSync('/dev/stdin').toString().trim().split(" ").map(Number);

let cnt = 0;
let answer = ''
const li = []
const recur = () => {
    if(li.length == k) {
        answer += li.join(" ") + "\n"
        return;
    }
    
    for(let i = 1; i <=n; i++){
        li.push(i);
        recur()
        li.pop();
    }
}

recur()

console.log(answer)